### PR TITLE
ci: setup Codeql workflow

### PR DESCRIPTION
**Pull request by mmorel-35**
Hi !

This is a setup for Codeql for kuma-gui as it is packaged in kuma.
I originally created an issue [in kuma](https://github.com/kumahq/kuma/issues/4646)
And a [pull request](https://github.com/kumahq/kuma/pull/5242).
The use of Codeql in this repository would help complete the coverage of the vulnerabilities detection :) 

Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by kleinfreund** [comment]
**Suggestion**: Please merge the latest changes from the `master` branch into your PR’s branch. I’ve made changes to the mocks.ts file that should avoid the reported CodeQL warnings.

**Comment by mmorel-35** [comment]
Hi @kleinfreund,

I rebased my branch, to integrate your latest modifications

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```
Comment (slonka) [inline .github/workflows/codeql.yml:23]: We have some `typescript` as well, right?

```suggestion
        language: [ 'javascript', 'typescript' ]
```

Comment (mmorel-35) [reply .github/workflows/codeql.yml:23]: I would think that Typescript is already scanned with javascript language, see https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/#languages-and-compilers 
I it saying:
> TypeScript analysis is performed by running the JavaScript extractor with TypeScript enabled. This is the default for LGTM.

**Review by slonka — APPROVED** [review]

**Comment by slonka** [comment]
Thanks for your contribution 🎉

**Pull request merged**